### PR TITLE
Use authentication with Docker Hub

### PR DESCRIPTION
**Pull request by xhlulu**
## Pre-Merge checklist
- [ ] The project was correctly built with `npm run build:all`.
- [ ] If there was any conflict, it was solved correctly.
- [ ] All changes were documented in CHANGELOG.md.
- [ ] All tests on CircleCI have passed.
- [ ] All Percy visual changes have been approved.
- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash Pivottable core contributor.

## Reference Issues

Closes #[issue number]

## Other comments

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ jobs:
     "node":
         docker:
             - image: circleci/python:3.7-node
+              auth:
+                  username: dashautomation
```
Comment (rpkyle) [inline .circleci/config.yml:8]: :trophy:

**Review by rpkyle — CHANGES_REQUESTED** [review]
Looks mostly OK, I think you need to add the context in your workflow though, so after `jobs:` you would add

```
context: dash-docker-hub
```

**Review by rpkyle — APPROVED** [review]
:checkered_flag: once tests pass! Thanks!

**Comment by rpkyle** [comment]
@xhlulu Sorry, I meant that it needs to be right after the job name below `jobs:`, see here:

image

File: .circleci/config.yml [hunk 2]
```diff
@@ -114,7 +114,9 @@ workflows:
     version: 2
     build:
         jobs:
-            context: dash-docker-hub
             - "python-3.6"
```
Comment (rpkyle) [inline .circleci/config.yml:117]: YAML can be a bit fiddly, but the colons are required (this happened to me as well).

```suggestion
            - "python-3.6":
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -114,7 +114,9 @@ workflows:
...
+                context: dash-docker-hub
             - "python-3.7"
```
Comment (rpkyle) [inline .circleci/config.yml:119]:
```suggestion
            - "python-3.7":
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -114,7 +114,9 @@ workflows:
...
+                context: dash-docker-hub
             - "node"
```
Comment (rpkyle) [inline .circleci/config.yml:121]:
```suggestion
            - "node":
```

**Comment by xhlulu** [comment]
Seems like it's running now

**Comment by rpkyle** [comment]
:woman_dancing:

**Comment by xhlulu** [comment]
Wow
```
tests/test_callbacks.py:27:12: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
Your code has been rated at 9.41/10
```

**Comment by xhlulu** [comment]
Seems like all the other tests passed except pylint, so might as well just merge

**Pull request merged**